src/main/java/com/commercetools/sync/commons/BaseSync.java

```diff
@@ -22,44 +26,54 @@ protected BaseSync(@Nonnull final U statistics, @Nonnull final V syncOptions) {
      * actions on the existing resource if it exists or create it if it doesn't.
      *
      * @param resourceDrafts the list of new resources as drafts.
-     * @return an instance of {@link CompletionStage&lt;U&gt;} which contains as a result an instance of {@link U} which
-     *      is a subclass of {@link BaseSyncStatistics} representing the {@code statistics} instance attribute of
-     *      {@link this} {@link BaseSync}.
+     * @return an instance of {@link CompletionStage&lt;U&gt;} which contains as a result an instance of {@link V} which
+     *      is a subclass of {@link BaseSyncStatisticsBuilder} representing the builder of {@code statistics} baked for
+     *      the sync process
      */
-    protected abstract CompletionStage<U> process(@Nonnull final List<T> resourceDrafts);
+    protected abstract CompletionStage<V> process(@Nonnull final List<T> resourceDrafts);
```

Comment:
please add more explanation in the doc why the builder is returned as a result from this process (to set the processing time).. And how is it used, would be useful for the devs using it.